fractal_server/app/runner/_slurm/__init__.py

```diff
@@ -0,0 +1,92 @@
+from pathlib import Path
+from typing import Any
+from typing import Dict
+from typing import List
+from typing import Optional
+
+from ...models import Workflow
+from .._common import recursive_task_submission
+from ..common import async_wrap
+from ..common import TaskParameters
+from .executor import FractalSlurmExecutor
+
+
+"""
+Slurm Bakend
+
+This backend runs fractal workflows as separate processes using a python
+thread process pool, where each thread is responsible for running a single
+task in a subprocess.
+
+Incidentally, it represents the reference implementation for a backend.
+"""
+
+
+def _process_workflow(
+    *,
+    workflow: Workflow,
+    input_paths: List[Path],
+    output_path: Path,
+    input_metadata: Dict[str, Any],
+    logger_name: str,
+    workflow_dir: Path,
+    username: str = None,
+    worker_init: Optional[
+        str
+    ] = None,  # this is only to match to _parsl interface
+) -> Dict[str, Any]:
+    """
+    TODO:
+    in case of failure we must return the most recent clean metadata
+
+    Returns:
+    output_dataset_metadata (Dict):
+        the output metadata
+    """
+
+    with FractalSlurmExecutor(
+        debug=True,
+        keep_logs=True,
+        additional_setup_lines=("#SBATCH --partition=main",),
```

Comment:
This should not be hard-coded, but it should depend on the task executor.

The same holds for other setup lines (namely the cpu/memory requirements), which should be configurable from `fractal_server.app.runner` according to the content of `WorkflowTask.meta` or `Task.meta`.

Reply:
deferred to #278